Apollo/Binary/Decoder.cs

```diff
@@ -564,6 +564,14 @@ public static class Decoder {
                     (RotateType)reader.ReadInt32(),
                     reader.ReadBoolean()
                 );
+
+            else if (t == typeof(Refresh)) {
+                bool[] macros;
+                macros = (from i in Enumerable.Range(0, 4) select reader.ReadBoolean()).ToArray();
```

Comment:
you don't need to do this explicitly unless there is a version check. this can go straight into the Refresh init (see example with Rotate above)